mteb/evaluation/evaluators/RetrievalEvaluator.py

```diff
@@ -154,15 +154,14 @@ def search(
                     cos_scores_top_k_idx[query_itr], cos_scores_top_k_values[query_itr]
                 ):
                     corpus_id = corpus_ids[corpus_start_idx + sub_corpus_id]
-                    if corpus_id != query_id:
-                        if len(result_heaps[query_id]) < top_k:
-                            # Push item on the heap
-                            heapq.heappush(result_heaps[query_id], (score, corpus_id))
-                        else:
-                            # If item is larger than the smallest in the heap, push it on the heap then pop the smallest element
-                            heapq.heappushpop(
-                                result_heaps[query_id], (score, corpus_id)
-                            )
+                    if len(result_heaps[query_id]) < top_k:
+                        # Push item on the heap
+                        heapq.heappush(result_heaps[query_id], (score, corpus_id))
+                    else:
+                        # If item is larger than the smallest in the heap, push it on the heap then pop the smallest element
+                        heapq.heappushpop(
+                            result_heaps[query_id], (score, corpus_id)
+                        )
```

Comment:
According to the previous discussion and my understanding, I think that we can't really remove this condition since it was introduced by BEIR for Quora & ArguAna that are included in the MTEB. 

In this case, I would say that it would be better to have a flag to activate/deactivate the self retrieval, so that the evaluator can also be used with MIRACL. This would mean that we would keep both scores as @KennethEnevoldsen suggested.

Else, if removing self retrieval and running again Quora and ArguAna makes sense then we should do it. Let me know what you think.